openmdao/core/group.py

```diff
@@ -1636,7 +1643,12 @@ def connect(self, src_name, tgt_name, src_indices=None, flat_src_indices=False):
             raise TypeError("%s: src_indices must be an index array, did you mean"
                             " connect('%s', %s)?" % (self.msginfo, src_name, tgt_name))
 
-        if isinstance(src_indices, Iterable):
+        if isinstance(src_indices, tuple):
+            contains_slice = [True if isinstance(i, slice) else False for i in src_indices]
+            if True not in contains_slice and isinstance(src_indices, Iterable):
```

Comment:
The `isinstance(src_indices, Iterable)` is unnecessary since you're inside of a conditional block where src_indices is a tuple.